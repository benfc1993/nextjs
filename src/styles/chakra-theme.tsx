import { theme, DefaultTheme, ThemeProvider } from '@chakra-ui/core'
import { render, RenderOptions } from '@testing-library/react'

import '@react-page/core/lib/index.css'

type ThemeProps = DefaultTheme & {
    sizes: any
    colors: any
    breakpoints: any
}

// [container, breakpoint]
const sizes = {
    sm: ['520px', '576px'],
    md: ['720px', '768px'],
    lg: ['960px', '992px'],
    xl: ['1170px', '1200px'],
}

export const breakpoints = Object.values(sizes).reduce((bp, size) => {
    bp.push(size[1])
    return bp
}, []) as any

export const containers = Object.values(sizes).reduce((bp, size) => {
    bp.push(size[0])
    return bp
}, []) as any

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

containers.sm = containers[0]
containers.md = containers[1]
containers.lg = containers[2]
containers.xl = containers[3]

export const Theme: ThemeProps = {
    ...theme,
    breakpoints,
    sizes: {
        ...theme.sizes,
        containers,
        breakpoints: Object.entries(sizes).reduce((bp, size) => {
            bp[size[0]] = size[1][1]
            bp[`${size[0]}Raw`] = Number.parseInt(size[1][1].replace('px', ''), 10)
            return bp
        }, {} as any),
    },
    colors: {
        ...theme.colors,
        brand: {
            red: '#DD0808',
            paleRed: '#FDF1F2',
            lightGrey: '#F2F2F2',
            grey: '#D8D8D8',
            darkGrey: '#505050',
            black: '#222222',
            white: '#FFFFFF',
            green: '#2DB200',
            paleGreen: '#E9F7E5',
        },
    },
    fonts: {
        ...theme.fonts,
        heading: 'Mulish, sans-serif',
        body: 'Mulish, sans-serif',
    },
}

// Custom helper to easily apply theme providers to components (for stories/testing etc.)
const withProviders: React.FC = ({ children }) => (
    <ThemeProvider theme={Theme}>{children}</ThemeProvider>
)
const customRender: any = (ui: React.ReactElement, options?: RenderOptions) =>
    render(ui, { wrapper: withProviders, ...options })

export { customRender as renderWithTheme }
export { withProviders as ThemeProvider }
