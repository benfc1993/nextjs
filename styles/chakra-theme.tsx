import React from 'react'
import { theme, DefaultTheme, ThemeProvider } from '@chakra-ui/core'

import '@react-page/core/lib/index.css'

type ThemeProps = DefaultTheme & {
    sizes: any
    colors: any
    breakpoints: any
    fonts: any
}

// [container, breakpoint]
const sizes = {
    sm: ['520px', '576px'],
    md: ['720px', '768px'],
    lg: ['960px', '992px'],
    xl: ['1170px', '1200px'],
}

export const breakpoints = Object.values(sizes).reduce((bp, size) => {
    bp.push(size[1])
    return bp
}, []) as any

export const containers = Object.values(sizes).reduce((bp, size) => {
    bp.push(size[0])
    return bp
}, []) as any

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

containers.sm = containers[0]
containers.md = containers[1]
containers.lg = containers[2]
containers.xl = containers[3]

export const Theme: ThemeProps = {
    ...theme,
    breakpoints,
    sizes: {
        ...theme.sizes,
        containers,
        breakpoints: Object.entries(sizes).reduce((bp, size) => {
            bp[size[0]] = size[1][1]
            bp[`${size[0]}Raw`] = Number.parseInt(size[1][1].replace('px', ''), 10)
            return bp
        }, {} as any),
    },
    colors: {
        ...theme.colors,
        brand: {
            green: '#5DBA23',
            lightgreen: '#EEF7E8',
            red: '#C20016',
            black: '#333333',
            grey: '#D8D8D8',
            darkgrey: '#D8D8D8',
            lightgrey: '#F2F2F2',
            white: '#FFFFFF',
        },
    },
    fonts: {
        ...theme.fonts,
    },
}

// Custom helper to easily apply theme providers to components (for stories/testing etc.)
const withProviders: React.FC = ({ children }) => (
    <ThemeProvider theme={Theme}>{children}</ThemeProvider>
)

export { withProviders as ThemeProvider }
