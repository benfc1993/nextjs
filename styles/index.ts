export * from './chakra-theme'
export * from './global'
