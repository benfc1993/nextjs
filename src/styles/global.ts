import { css } from '@emotion/core'
import { Theme } from './chakra-theme'

import 'typeface-muli'

// Global styles designs: link to the client's style guide if there is one.

const listStyles = css`
    ul {
        list-style: none;
        margin-left: 0;
        margin-right: 0;
        margin-bottom: 20px;
        position: relative;
        li {
            line-height: 30px;
            padding-left: 20px;
            position: relative;
            &:before {
                display: block;
                content: '';
                position: absolute;
                left: 0;
                top: 10px;
                height: 8px;
                width: 8px;
                border-radius: 100%;
                margin-right: 10px;
                background-color: ${Theme.colors.brand.black};
            }
        }
    }
`

const headings = css`
    h1 {
        font-size: 30px;
        line-height: 36px;
        margin-bottom: 10px;

        @media (max-width: 478px) {
            font-size: 24px;
            line-height: 30px;
        }
    }

    h2 {
        font-size: 28px;
        line-height: 34px;
        margin-bottom: 15px;

        @media (max-width: 478px) {
            font-size: 22px;
            line-height: 28px;
        }
    }

    h3 {
        font-size: 24px;
        line-height: 30px;

        @media (max-width: 478px) {
            font-size: 20px;
            line-height: 26px;
        }
    }

    h4 {
        font-size: 18px;
        line-height: 24px;
    }

    h5 {
        font-size: 16px;
        line-height: 22px;
    }

    h6 {
        font-size: 14px;
        line-height: 20px;
    }
`

export const globalStyles = css`
    ${listStyles}
    ${headings}
`
