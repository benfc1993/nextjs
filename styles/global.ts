import { css } from '@emotion/core'
import { Theme } from './chakra-theme'

const inputs = css`
    input:-internal-autofill-selected {
        background-color: red !important;
    }
    input::placeholder {
        color: ${Theme.colors.darkGrey} !important;
        position: relative;
        top: -3px;
    }

    select,
    input[type='text'] {
        &:disabled {
            background: ${Theme.colors.brand.lightgrey};
            color: #ccc;

            & + div {
                color: #ccc;
            }
        }
    }

    div[type='radio'] {
        background-color: ${Theme.colors.brand.white};
    }

    button:focus {
        box-shadow: none !important;
    }
`

const unorderedList = css`
    ul {
        list-style: none;
        margin-left: 0;
        margin-right: 0;
        margin-bottom: 20px;
        position: relative;
        li {
            line-height: 30px;
            padding-left: 20px;
            position: relative;
            &:before {
                display: block;
                content: '';
                position: absolute;
                left: 0;
                top: 10px;
                height: 10px;
                width: 10px;
                margin-right: 10px;
                border-radius: 5px;
                background-color: ${Theme.colors.brand.black};
            }
            ul {
                margin-left: 10px;
            }
        }
    }
`

const carousel = css`
    .carousel {
        ul {
            margin-bottom: 0;
            li {
                padding-left: 0;
                &:before {
                    content: none;
                }
            }
        }
    }

    .carousel .slide,
    .DayPicker {
        text-align: left;
        background: transparent;

        ul {
            margin-bottom: 0;
            li {
                padding-left: 0;
                &:before {
                    content: none;
                }
            }
        }
    }

    .carousel.carousel-slider {
        overflow: visible;
        display: flex;
        flex-direction: row;
        align-items: center;
    }

    .control-dots {
        display: flex;
        justify-content: center;
    }
`

const sharedHeadingsStyles = css`
    font-family: 'Rooster', sans-serif;
    font-weight: bold;
`

const headings = css`
    h1 {
        ${sharedHeadingsStyles};

        @media (max-width: 478px) {
            font-size: 24px;
            line-height: 30px;
        }

        font-size: 40px;
        font-weight: 500;
        line-height: 48px;
        margin-bottom: 10px;
    }

    h2 {
        ${sharedHeadingsStyles};

        @media (max-width: 478px) {
            font-size: 22px;
            line-height: 28px;
        }

        font-size: 24px;
        font-weight: 500;
        line-height: 30px;
        margin-bottom: 15px;
    }

    h3 {
        ${sharedHeadingsStyles};

        @media (max-width: 478px) {
            font-size: 18px;
            line-height: 24px;
        }

        font-size: 20px;
        font-weight: 500;
        line-height: 26px;
    }

    h4 {
        ${sharedHeadingsStyles};
        font-size: 18px;
        font-weight: 500;
        line-height: 24px;
    }

    h5 {
        ${sharedHeadingsStyles};
        font-size: 18px;
        font-weight: 500;
        line-height: 24px;
    }

    h6 {
        ${sharedHeadingsStyles};
        font-size: 14px;
        font-weight: 500;
        line-height: 20px;
    }
`

const componentWrapperSpecific = css`
    .component-wrapper {
        p {
            margin-bottom: 15px;
        }

        a {
            text-decoration: underline;
        }

        img {
            align-self: center;
        }

        > div > div {
            > :firtst-of-type {
                margin-top: 0;
            }

            > :last-child {
                margin-bottom: 0px;
            }
        }

        table {
            border-collapse: collapse;

            td,
            th {
                padding: 0.75rem;
                vertical-align: top;
                border-top: 1px solid #dee2e6;
            }
        }
        textarea {
            width: 100%;
            display: flex;
            -webkit-box-align: center;
            align-items: center;
            position: relative;
            transition: all 0.2s ease 0s;
            outline: none;
            appearance: none;
            font-size: 14px;
            padding-left: 15px;
            padding-right: 15px;
            height: 45px;
            border-radius: 0px !important;
            border: 1px solid ${Theme.colors.brand.lightgrey};
            background-color: rgb(255, 255, 255);
            padding-top: 10px;
            &:hover,
            &:focus {
                border-color: #5dba23 !important;
            }
        }

        div[type='radio'] {
            background-color: ${Theme.colors.brand.white};
        }
    }
`

const links = css`
    a {
        :hover {
            text-decoration: underline;
            text-decoration-color: inherit;
        }
        :focus {
            box-shadow: none !important;
        }
    }
`

export const globalStyles = css`
    html,
    body {
        padding: 0;
        margin: 0;
        font-family: 'Rooster', sans-serif;
    }
    ul {
        list-style: none;
        margin-left: 0;
        margin-right: 0;
        margin-bottom: 20px;
        position: relative;
        li {
            line-height: 30px;
            padding-left: 20px;
            position: relative;
            &:before {
                display: block;
                content: '';
                position: absolute;
                left: 0;
                top: 10px;
                height: 10px;
                width: 10px;
                margin-right: 10px;
                background-color: ${Theme.colors.brand.black};
            }
        }
    }
    .sb-show-main {
        background-color: #f2f2f2;
    }
    ${inputs}
    ${unorderedList}
    ${carousel}
    ${headings}
    ${links}
    ${componentWrapperSpecific}
`
