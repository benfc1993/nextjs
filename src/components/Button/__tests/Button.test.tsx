import React from 'react'
import { Button } from '..'
import { renderWithTheme } from '../../../styles/chakra-theme'

describe('Button', () => {
    it('matches the previous snapshot', () => {
        const { asFragment } = renderWithTheme(<Button>Button Text</Button>)
        expect(asFragment()).toMatchSnapshot()
    })
})
