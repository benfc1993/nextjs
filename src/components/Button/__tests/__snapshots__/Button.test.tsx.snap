// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches the previous snapshot 1`] = `
<DocumentFragment>
  <button
    class="css-164j1l2"
    type="button"
  >
    Button Text
  </button>
</DocumentFragment>
`;
